packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -0,0 +1,198 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  queries,
+  within,
+  type BoundFunctions,
+  type ByRoleOptions,
+  type GetByText,
+  GetByRole,
+} from "@testing-library/dom";
+
+import {
+  type FeatureTestSelector,
+  type TestSelector,
+} from "./featureTestSelector";
+import { odysseyTestSelectors } from "./odysseyTestSelectors";
+
+export const interpolateString = (
+  string: string,
+  values: Record<string, string | RegExp>,
+) => {
+  const interpolatedString = eval(`
+    ${Object.entries(values)
+      .map(
+        ([key, value]) =>
+          `const ${key} = ${
+            typeof value === "string" ? JSON.stringify(value) : value
+          };`,
+      )
+      .join("")}
+
+    \`${string}\`
+  `) as string;
+
+  if (/^\/*(.+)\/$/.test(interpolatedString)) {
+    return eval(interpolatedString) as RegExp;
+  }
+
+  return interpolatedString;
+};
+
+const getByQuerySelector = ({
+  canvas,
+  method,
+  options,
+  role,
+  text,
+}: {
+  canvas: BoundFunctions<typeof queries>;
+  method: "ByRole" | "ByLabelText" | "ByPlaceholderText" | "ByText";
+  options?: ByRoleOptions;
+  role?: Parameters<GetByRole>[1];
+  text?: Parameters<GetByText>[1];
+}) => {
+  if (method === "ByRole") {
+    return canvas.getByRole(
+      // TODO: These should eventually reference `query` as the function identifier.
+      role!,
+      options,
+    );
+  } else if (method === "ByLabelText") {
+    return canvas.getByLabelText(
+      // These should eventually reference `query` as the function identifier.
+      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
+      options,
+    );
+  } else if (method === "ByPlaceholderText") {
+    return canvas.getByPlaceholderText(
+      // These should eventually reference `query` as the function identifier.
+      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
+      options,
+    );
+  } else if (method === "ByText") {
+    return canvas.getByText(
+      // These should eventually reference `query` as the function identifier.
+      text!, // TODO: Use TypeScript `Infer` to ensure `description` is required when it's `ByLabelText`.
```

Comment:
Anything we need to do about these TODO's?

Reply:
Given more time, I could go back and fix them. They're a bit complicated to fix with TS without spending another day on it.

I'd like to leave this for a follow-up PR that also includes typing the returned element more specifically. Instead of just `HTMLElement`, it types it as `HTMLInputElement`.

What we have today works fine with the 3 exposed components. As we add more, we can refine this as necessary.

Reply:
Yea sounds good. Just wanted to make sure we didn't miss something here. Thanks.